qutebrowser/keyinput/modeman.py

```diff
@@ -294,6 +294,16 @@ def leave(self, mode, reason=None, maybe=False):
 
         log.modes.debug("Leaving mode {}{}".format(
             mode, '' if reason is None else ' (reason: {})'.format(reason)))
+
+        if mode == usertypes.KeyMode.insert and \
+           config.val.input.blur_on_mode_leave:
+
+            command_dispatcher = objreg.get('command-dispatcher',
```

Comment:
This is a bit of a hack, is there a better way to get a `jseval` into here, or is this the best way?

Reply:
You probably shouldn't do this in `modeman.py` - instead, add something like `_on_mode_left` in `AbstractCaret` in `qutebrowser/browser/browsertab.py` to some other place where it's convenient and where you have access to the current `tab` object, and call its method to evaluate the JS.